File: src/de/schauderhaft/cleanCode/exercise6/TextStrategy.java
package de.schauderhaft.cleanCode.exercise6;

import de.schauderhaft.cleanCode.stubs.SrvParameterCache;
import de.schauderhaft.cleanCode.stubs.StringStatic;
import de.schauderhaft.cleanCode.stubs.Translator;

public class TextStrategy {
    InstanceClassifier classifier = new InstanceClassifier();

    String getText() {

        String text = Translator.translate("lblStageInfoPrd");

        String importUrl = SrvParameterCache
                .getParameterAsString(SrvParameterCache.WEB_URL_DEFAULT);
        if (!StringStatic.isEmptyOrNullValue(importUrl)) {
            importUrl = importUrl.toLowerCase();

            if (classifier.isTestInstance(importUrl)) {
                text = Translator.translate("lblStageInfoTest");
            } else if (classifier.isQaInstance(importUrl)) {
                text = Translator.translate("lblStageInfoQs");
            }
        }

        return text;
    }
}

File: src/de/schauderhaft/cleanCode/exercise6/TextStrategy.cs
using de.schauderhaft.cleanCode.stubs;

namespace de.schauderhaft.cleanCode.exercise6
{
    public class TextStrategy
    {
        private InstanceClassifier classifier = new InstanceClassifier();
        
        public string GetText()
        {
            if (classifier.IsTestInstance()) 
            {
                return Translator.Translate("lblStageInfoTest");
            }
            else if (classifier.IsQaInstance()) 
            {
                return Translator.Translate("lblStageInfoQs");
            }
            else
            {
                return Translator.Translate("lblStageInfoPrd");
            }
        }
    }
}